google/cloud/aiplatform/models.py

```diff
@@ -3503,3 +3848,309 @@ def get_model_evaluation(
                 evaluation_name=evaluation_resource_name,
                 credentials=self.credentials,
             )
+
+
+# TODO (b/232546878): Async support
+class ModelRegistry:
+    def __init__(
+        self,
+        model: Union[Model, str],
+        location: Optional[str] = None,
+        project: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> None:
+        """Creates a ModelRegistry instance for version management of a registered model.
+
+        Args:
+            model (Union[Model, str]):
+                Required. One of the following:
+                    1. A Model instance
+                    2. A fully-qualified model resource name
+                    3. A model ID. A location and project must be provided.
+            location (str):
+                Optional. The model location. Used when passing a model name as model.
+                If not set, project set in aiplatform.init will be used.
+            project (str):
+                Optional. The model project. Used when passing a model name as model.
+                If not set, project set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use with model access. If not set,
+                credentials set in aiplatform.init will be used.
+        """
+
+        if isinstance(model, Model):
+            self.model_resource_name = model.resource_name
+        else:
+            self.model_resource_name = utils.full_resource_name(
+                resource_name=model,
+                resource_noun="models",
+                parse_resource_name_method=Model._parse_resource_name,
+                format_resource_name_method=Model._format_resource_name,
+                project=project,
+                location=location,
+                resource_id_validator=base.VertexAiResourceNoun._revisioned_resource_id_validator,
+            )
+
+        self.credentials = credentials or (
+            model.credentials if isinstance(model, Model) else None
+        )
+        self.client = Model._instantiate_client(location, self.credentials)
```

Comment:
Resolved. Instead of self.credentials being None, self.credentials will be set to the default credentials. Models retrieved from the registry will use the saved self.credentials, even if the default creds change after MR creation.

Reply:
LGTM